iothub/service/src/Authentication/EventHubConnectionStringBuilder.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Amqp.Sasl;
+using Microsoft.Azure.Amqp.Transport;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Gets an event hub-compatible connection string to be used with <see href="https://www.nuget.org/packages/Azure.Messaging.EventHubs"/>.
```

Comment:
Copied this (and updated) from mono repo samples.

We've been asked to integrate this into our SDK for quite a while. Long story, EventHubs SDK used to have a ctor that would take an IoT hub connection string. They removed it after a major rev because it was odd an service reverse dependency.